cli/tests/containerize.bats

```diff
@@ -43,19 +43,109 @@ podman_cache_reset() {
+    export HOME="$REAL_HOME"
+  else
+    tmpdir="$(mktemp -d "/tmp/home.XXXXXX")"
+    mkdir -p "$tmpdir"
+    export FLOX_TEST_HOME="$tmpdir"
+    # Force recreation on `home' on every invocation.
+    unset __FT_RAN_HOME_SETUP
+  fi
+  echo "Podman home dir: $FLOX_TEST_HOME" >&3
+  export __FT_RAN_HOME_SETUP="$FLOX_TEST_HOME"
+}
+
+podman_xdg_vars_setup() {
+  home_dir="$1"; shift;
+  short_tmp_dir="$1"; shift;
+
+  xdg_reals_setup
+  # These get unset by the preceding function call and must be restored in order
+  # to use a single podman machine across the test run
+  export XDG_CONFIG_HOME="$short_tmp_dir/.config"
+  export XDG_DATA_HOME="$short_tmp_dir/.local/share"
+  export XDG_RUNTIME_DIR="$short_tmp_dir/run"
+
+  test_cache_dir="${home_dir:?}/.cache"
+  test_state_dir="${home_dir:?}/.local/state"
+
+  # Create all of the directories
+  mkdir -p "$home_dir"
+  mkdir -p "$test_cache_dir"
+  mkdir -p "$test_state_dir"
+  chmod u+w "$home_dir"
+  chmod u+w "$test_cache_dir"
+  chmod u+w "$test_state_dir"
+
+  # Export the vars
+  export XDG_CACHE_HOME="$test_cache_dir"
+  export XDG_STATE_HOME="$test_state_dir"
+}
+
+# This is the same as the global `flox_vars_setup` except it doesn't run
+# `xdg_vars_setup` again.
+podman_flox_vars_setup() {
+  # We store sockets in FLOX_CACHE_DIR,
+  # so create cache in /tmp since TMPDIR may result in too long of a path.
+  FLOX_CACHE_DIR="$(mktemp -d /tmp/flox.tests.XXXXXX)"
+  export FLOX_CACHE_DIR
+  export FLOX_CONFIG_DIR="$XDG_CONFIG_HOME/flox"
+  export FLOX_DATA_HOME="$XDG_DATA_HOME/flox"
+  export FLOX_STATE_HOME="$XDG_STATE_HOME/flox"
+  export FLOX_META="$FLOX_CACHE_DIR/meta"
+  export FLOX_ENVIRONMENTS="$FLOX_DATA_HOME/environments"
+  export HOME="${FLOX_TEST_HOME:-$HOME}"
+}
+
+
+podman_dirs_setup() {
+  # Podman creates deeply nested directories and stores sockets in some of them,
+  # so we need to create locations to store those with shorter paths than what
+  # we'd get nesting them under `/tmp/nix-shell.XXXXXX/bats-run-XXXXXX`.
+  export SHORT_TMP="$(mktemp -d "/tmp/XXXXXX")"
+  export TMPDIR="$SHORT_TMP"
+  export XDG_CONFIG_HOME="$SHORT_TMP/.config"
+  export XDG_DATA_HOME="$SHORT_TMP/.local/share"
+  export XDG_RUNTIME_DIR="$SHORT_TMP/run"
+  echo "Podman XDG root: $SHORT_TMP" >&3
+  mkdir -p "$XDG_CONFIG_HOME/containers"
+  mkdir -p "$XDG_DATA_HOME"
+  mkdir -p "$XDG_RUNTIME_DIR"
+
+  # Don't require Rosetta unless cross-compiling on builders.
+  if [[ "${NIX_SYSTEM}" != "x86_64-darwin" ]]; then
+    cat > "$XDG_CONFIG_HOME/containers/containers.conf" <<'EOF'
+[machine]
+rosetta = false
+EOF
+  fi
+  
+  podman_home_setup
+  podman_xdg_vars_setup "$FLOX_TEST_HOME" "$SHORT_TMP"
+  podman_flox_vars_setup
+}
+
+start_podman_machine() {
+  machine="$(podman machine list -n)"
+  if [ -z "$machine" ]; then
+    echo "Creating podman machine" >&3
+    podman machine init -v /tmp:/tmp -v /Users:/Users -v /private:/private
+  fi
+  echo "Starting podman machine" >&3
+  podman machine start
```

Comment:
It's probably worth keeping the `--log-level=debug` because it has been the only clue about the 2~3 different ways that it's failed so far and the output is hidden when the test passes.

Reply:
Done

Reply:
Actually, this causes the job to hang, so it's something I'd prefer we enable on an ad-hoc basis until we figure out a workaround.